src/EditorFeatures/Core/Shared/Tagging/Utilities/TagSpanIntervalTree.cs

```diff
@@ -57,10 +57,10 @@ private static TagSpan<TTag> GetTranslatedTagSpan(TagSpan<TTag> originalTagSpan,
             : new(GetTranslatedSpan(originalTagSpan, textSnapshot, trackingMode), originalTagSpan.Tag);
 
     public bool HasSpanThatContains(SnapshotPoint point)
-        => _tree.HasIntervalThatContains(point.Position, length: 0, new IntervalIntrospector(point.Snapshot, _spanTrackingMode));
+        => _tree.Algorithms.HasIntervalThatContains(point.Position, length: 0, new IntervalIntrospector(point.Snapshot, _spanTrackingMode));
```

Comment:
these helpers could not be extensions for two reasons:

1. the types themselves have some existing intance methods that have hte same name.  and we never look up extensions if we find a matching instance name (even if it fails).
2. The extensions were on a `TIntervalTree` so that they can operate without calling through an interface once instantiated.  but the inference required there is beyond what C# can do today.  So i used the pattern of asking the instance for a wrapper struct that itself then exposes the operations.  The property handles the initial part of generic inference, meaning that the only inference has to happen on the TIntrospector type, which works fine.